sensor/kubernetes/certrefresh/certificates/requester.go

```diff
@@ -187,37 +190,45 @@ func (r *genericRequester[ReqT, RespT]) RequestCertificates(ctx context.Context)
 		}
 	}
 
+	// create a new channel for this specific request, and store it in the requests map
 	requestID := uuid.NewV4().String()
-	receiveC := make(chan RespT, 1)
-	r.requests.Store(requestID, receiveC)
-	defer r.requests.Delete(requestID)
+	certsResponseC := make(chan RespT, 1)
+	r.requests.Store(requestID, certsResponseC)
+
 	// Always delete this entry when leaving this scope to account for requests that are never responded, to avoid
 	// having entries in `r.requests` that are never removed.
+	defer r.requests.Delete(requestID)
+
 	if err := r.send(ctx, requestID); err != nil {
 		return nil, err
 	}
```

Comment:
> It looks like this function does not need the channel at all, because it always sends 0 or 1 messages per call. Why do we use the channel here then?

Are you referring to the `certsResponseC` channel?

Reply:
I'm not sure why it was implemented like this because I did not write this, but it looks simpler than alternatives I could think of (but then again, wouldn't work with a blocking channel).

Another point worth mentioning is that `RequestCertificates` supports concurrent calls, but this is not needed and not used (the function is always called from the same thread and it's blocking). I could remove this requirement and simplify the logic. WDYT?

Reply:
I looks like this would be more effective to do together in a call, as this requires me to go deeper into the logic of this part. I wanted to avoid this and only review the sensor-relevant parts but that may be difficult. 

> I'm not sure why it was implemented like this because I did not write this, but it looks simpler than alternatives I could think of (but then again, wouldn't work with a blocking channel).

I apologize, but I cannot accept this as a valid reason to leave it like this. Copying code that is bad and leaving it unrefactored causes that all future problems will be coming back to the Maple team. There is only one way to prevent that and this is what I am trying to convince you to do by reviewing this PR and applying changes to the code that was written by different people in the past.

Reply:
I didn't say that I want to leave it as is. But as an alternative to refactoring it, I was proposing to simplify the logic, because `RequestCertificates` doesn't really have to support concurrent calls.